api/s2n.h

```diff
@@ -2291,7 +2291,26 @@ S2N_API extern int s2n_shutdown_send(struct s2n_connection *conn, s2n_blocked_st
 /**
  * Used to declare what type of client certificate authentication to use.
  *
- * Currently the default for s2n-tls is for neither the server side or the client side to use Client (aka Mutual) authentication.
+ * A s2n_connection will enforce client certificate authentication (mTLS) differently based on
+ * the s2n_cert_auth_type and s2n_mode(client/server) of the connection, as described below. The
+ * default behavior is used if the application hasn't set an override via
+ * `s2n_config_set_client_auth_type()` or `s2n_connection_set_client_auth_type()`.
+ *
+ * **Server behavior:**
+ * - None(default): don't request client authentication. Abort the handshake if the client
+ *      sends its certificate.
+ * - Optional: request the client's certificate and validate if it's non-empty. Abort the
+ *      handshake if the client doesn't send its certificate (can be empty).
```

Comment:
Dont love the "(can be empty)" and technically its redundant since i mention non-empty. Thoughts on if we should keep it or re-word it?